manager/shard_strategy.go

```diff
@@ -0,0 +1,226 @@
+package manager
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"hash/fnv"
+
+	"github.com/flyteorg/flytepropeller/manager/config"
+	"github.com/flyteorg/flytepropeller/pkg/apis/flyteworkflow/v1alpha1"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+// Defines necessary functionality for a sharding strategy
```

Comment:
```suggestion
// ShardStrategy defines necessary functionality for a sharding strategy
```
the style is to write the name of the struct followed by the sentence. Golint will complain otherwise

Reply:
done